svgelements/svgelements.py

```diff
@@ -5515,15 +5518,13 @@ def _validate_move(self, index):
 
     def _validate_close(self, index):
         """ensure the close element at this position correctly links to the previous move"""
-        for i in range(index, -1, -1):
+        for i in range(index - 1, -1, -1):
             segment = self._segments[i]
-            if isinstance(segment, Move):
+            if isinstance(segment, (Move, Close)):
                 self._segments[index].end = Point(segment.end)
```

Comment:
Even the _validate close says ensure close element correctly links with previous move. Not previous move or previous Close